### ISSUE TITLE
Crash in debug mode on windows VC2017

### ISSUE THREAD

**Issue by UnrealDoggie** [stevinz/extrude#2]
Hi,
When I run the compiled program, it crashes in Debug mode with assert some
vector subscript out of range.

In imaging_compare.cpp, function FloodFill

for (auto point : points) {

    // somewhere wherepush_back is done conditionally.
    if (processed.getPixel(x, y) == FLOOD_NOT_PROCESSED) {
                        if (Dr::IsSameColor(start_color, bitmap.getPixel(x, y), tolerance)) {
                            points.push_back(DrPoint(x, y));
                            processed.setPixel(x, y, FLOOD_MARKED_FOR_PROCESS);
                            processed_some = true;
                        }
    }
}

push_back will invalidate the iterators if there is reallocate of points
I added points.reserve(500) to prevent the reallocation & it ran correctly.
Alternatively a while loop with accessing points by index instead.

P.S: Is there any version of  Visual Studio / c++ should I be using?
My compiler does not seem to recognize  .path initialization etc.
I had to changed to that below.
/* sfetch_request_t (sokol_fetch_font) {
        .path = font_file.c_str(),
        .callback = font_normal_loaded,
        .buffer_ptr = state.font_normal_data,
        .buffer_size = sizeof(state.font_normal_data)
};*/
    sfetch_request_t sokol_fetch_font = { 0 };
    sokol_fetch_font.path = font_file.c_str();
    sokol_fetch_font.callback = font_normal_loaded;
    sokol_fetch_font.buffer_ptr = state.font_normal_data;
    sokol_fetch_font.buffer_size = sizeof(state.font_normal_data);

Regards,
Eugene

**Comment by stevinz** [comment]
Thanks for the feedback! Great catch on the invalidated iterator! I have updated the FloodFill function.

Some of the Sokol structs were being initialized using a newer syntax officially implemented in c++20 (https://en.cppreference.com/w/cpp/language/aggregate_initialization ... see 3 & 4 in Syntax table). I've been building on a mac with gcc and never had a problem compiling, even though I'm targeting c++11. Seems different compilers have added this at different times. I have changed all instances for better compatibility. 

I fired up my old Windows 10 machine and installed visual studio 19 community to check for other issues. There were a couple but it should now be building and running successfully on Windows.

**Comment by UnrealDoggie** [comment]
Ah ... c++20.

I am stuck with Visual Studio 2017.
Had to make some changes.

1) //added cklee for std::isdigit in string.h
 #include <cctype>
2) //added cklee remove static causes compiler out of heap space
//static state_t state;
state_t state;

I didnt really dwell deeper into the cause as my interest lies with the result of the extrusion as I am writing out in obj.
Is it possible to isolate the different objects as separate meshes?

**Comment by stevinz** [comment]
Okay, I downloaded Visual Studio 2017 (just grabbed the latest build, 15.9) to check for any problems. I resolved a couple issues and everything is building with cmake and running nice.

I also made changes to the DrMesh class so that meshes are created separately from the different objects found within the image. Now, in the demo, as the objects are processed, the separate meshes are stored in a std::vector<DrMeshes> called 'meshes'.

As far as creating obj files, that is currently out of the scope of this demo. But, in case you did try to to export the objects yourself, I did add a simple feature to adjust the depth. It could be helpful before you export. The depth of the objects can be adjusted while the demo is running by pressing the -/+ keys (also note, the quality of the object can be adjusted using the number keys).
